docs/topics/multiplatform/multiplatform-run-tests.md

```diff
@@ -1,67 +1,375 @@
+            return result?.value ?: "unknown"
+        }
+    }
+    ```
+
+3. In `commonTest`, create the new `org.kmp.testing/CurrentRuntimeTest.kt` file.
+4. Update it with a platform and framework-agnostic test like this:
 
-class SampleTestsIOS {
-    @Test
-    fun testHello() {
-        assertTrue("iOS" in hello())
+    ```kotlin
+    import kotlin.test.Test
+    import kotlin.test.assertEquals
+
+    class CurrentRuntimeTest {
+        @Test
+        fun shouldDisplayDetails() {
+            val runtime = CurrentRuntime("MyRuntime", "1.1")
+            assertEquals("MyRuntime version 1.1", runtime.toString())
+        }
+    
+        @Test
+        fun shouldHandleNullVersion() {
+            val runtime = CurrentRuntime("MyRuntime", null)
+            assertEquals("MyRuntime version unknown", runtime.toString())
+        }
+    
+        @Test
+        fun shouldParseNumberFromVersionString() {
+            val runtime = CurrentRuntime("MyRuntime", "1.2 Alpha Experimental")
+            assertEquals("MyRuntime version 1.2", runtime.toString())
+        }
+    
+        @Test
+        fun shouldHandleMissingVersion() {
+            val runtime = CurrentRuntime("MyRuntime", "Alpha Experimental")
+            assertEquals("MyRuntime version unknown", runtime.toString())
+        }
     }
-}
+    ```
+
+You can run this test using any of the ways [available in the IDE](#run-tests).
+
+### Writing platform-specific tests
+
+> Here, the mechanism of expected and actual declarations is used for brevity and simplicity. In more complex code, a
+> better approach is to use interfaces and factory functions.
+>
+{type="tip"}
+
+To create an instance of `CurrentRuntime`, declare a function in the common `CurrentRuntime.kt` file as follows:
+
+```kotlin
+expect fun determineCurrentRuntime(): CurrentRuntime
 ```
 
-You can also learn how to create and run multiplatform tests in the [Create and publish a multiplatform library – tutorial](multiplatform-library.md#test-your-library).
\ No newline at end of file
+The function should have separate implementations for each supported platform. Otherwise, the build would fail.
+
+As well as implementing this function on each platform, you should provide tests. Let's see how to do it both on Android
+and iOS.
+
+#### For Android
+
+1. In `androidMain`, create a new `org.kmp.testing/AndroidRuntime.kt` file.
+2. Add the actual implementation of the expected function:
+
+    ```kotlin
+    actual fun determineCurrentRuntime(): CurrentRuntime {
+        val name = System.getProperty("java.vm.name") ?: "Android"
+    
+        val version = System.getProperty("java.version")
+    
+        return CurrentRuntime(name, version)
+    }
+    ```
+
+3 Use the IDE's suggestions to create the `androidUnitTest/kotlin` directory:
+
+![Creating Android test directory](create-android-test-dir.png){width=300}
+
+4. Add the `AndroidRuntimeTest.kt` file to the project:
```

Comment:
I'm not sure where the `AndroidRuntimeTest.kt` file goes? In the `androidUnitTest/kotlin` directory?

Reply:
Yes, definitely needs some clarification then